### PR TITLE
[RW-3771][risk=no] Move help tips sidebar into workspace wrapper

### PR DESCRIPTION
**Pull request by dolbeew**
- Removes `HelpSidebar` from individual pages and adds it to `WorkspaceWrapperComponent` so it shows on all pages within a workspace (except `InteractiveNotebookComponent`).
- Adds a `helpContent` property to the data for each route with the help sidebar that determines the help tip content.
- Moves handling of participants/annotations for `SidebarContent` component into `review-state.service.ts` instead of using `props`.

@freemabd to deploy and test locally, especially annotations
@NehaBroad for general review, especially changes to workbench code

### CODE REVIEW

File: ui/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -131,15 +135,16 @@ const routes: Routes = [
...
+                      helpContent: 'data'
                     }
                   }, {
                     path: 'preview/:nbName',
                     component: InteractiveNotebookComponent,
                     data: {
                       pathElementForTitle: 'nbName',
                       breadcrumb: BreadcrumbType.Notebook,
-                      minimizeChrome: true
+                      minimizeChrome: true,
```
Comment (blrubenstein) [inline ui/src/app/app-routing.module.ts]: I think this is an unrelated change? Should revert unless we're planning to have the helpContent here? (Which may make sense. I don't know what the mocks look like.)

File: ui/src/app/components/help-sidebar.spec.tsx [hunk 2]
```diff
@@ -1,11 +1,26 @@
...
+  beforeEach(() => {
+    registerApiClient(CohortReviewApi, new CohortReviewServiceStub());
+    registerApiClient(CohortAnnotationDefinitionApi, new CohortAnnotationDefinitionServiceStub());
+    currentWorkspaceStore.next(workspaceDataStub);
+    cohortReviewStore.next(cohortReviewStubs[0]);
+  });
+
   it('should render', () => {
-    const wrapper = mount(<HelpSidebar location='data' />);
+    const wrapper = mount(<HelpSidebar helpContent='data' hideSidebar={0} />);
```
Comment (blrubenstein) [inline ui/src/app/components/help-sidebar.spec.tsx]: Can we add another test that checks whether the content changes from page to page?

Also, we should test that the sidebar is hidden if hide is set to true.

File: ui/src/app/components/help-sidebar.tsx [hunk 3]
```diff
@@ -124,15 +125,15 @@ const iconStyles = {
 };
 
 interface Props {
-  location: string;
+  helpContent: string;
+  hideSidebar: number;
```
Comment (blrubenstein) [inline ui/src/app/components/help-sidebar.tsx]: This feels like it should semantically be a number. It looks like it gets incremented on navigate? Could we instead set it to true on every navigate? That feels more like what we are looking to do here.

File: ui/src/app/components/help-sidebar.tsx [hunk 4]
```diff
@@ -124,15 +125,15 @@ const iconStyles = {
...
   profileState: any;
-  participant?: any;
-  setParticipant?: Function;
 }
 
 interface State {
   activeIcon: string;
   filteredContent: any;
+  participant: any;
```
Comment (blrubenstein) [inline ui/src/app/components/help-sidebar.tsx]: Nit: We should type this if possible. Maybe as `ParticipantCohortStatus` if I am understanding correctly? (And the filtered content)
Do we know what data it will have?

File: ui/src/app/components/help-sidebar.tsx [hunk 5]
```diff
@@ -156,6 +158,19 @@ export const HelpSidebar = withUserProfile()(
...
     });
 
+    componentDidMount(): void {
+      participantStore.subscribe(participant => this.setState({participant}));
+    }
+
+    componentDidUpdate(prevProps: Readonly<Props>): void {
+      const {helpContent, hideSidebar} = this.props;
+      // close the sidebar on each navigation excluding navigating between participants in cohort review
+      if (hideSidebar > prevProps.hideSidebar &&
```
Comment (blrubenstein) [inline ui/src/app/components/help-sidebar.tsx]: If this is a boolean, we could probably do this as: `hideSidebar && !prevProps.hideSidebar`, if I am understanding this correctly?

File: ui/src/app/components/help-sidebar.tsx [hunk 6]
```diff
@@ -156,6 +158,19 @@ export const HelpSidebar = withUserProfile()(
...
+        !(helpContent === 'reviewParticipantDetail' && prevProps.helpContent === 'reviewParticipantDetail')) {
+        this.setState({activeIcon: undefined, sidebarOpen: false});
```
Comment (blrubenstein) [inline ui/src/app/components/help-sidebar.tsx]: If we have `hideSidebar` as a boolean, could we get rid of `sidebarOpen`? It seems redundant? I am a little confused by the interaction of `sidebarOpen` and `hideSidebar` here. It seems to me like `sidebarOpen` is the source of truth, but parent components also need to manage state? Which suggests maybe we should lift the state up and use `sidebarOpen` to trigger a callback?

File: ui/src/app/pages/data/cohort-review/sidebar-content.component.tsx [hunk 7]
```diff
@@ -294,6 +294,7 @@ export const SidebarContent = fp.flow(
       .then(({items}) => {
         this.setState({annotationDefinitions: items});
       });
+    participantStore.subscribe(participant => this.setState({participant: participant || {} as ParticipantCohortStatus}));
```
Comment (blrubenstein) [inline ui/src/app/pages/data/cohort-review/sidebar-content.component.tsx]: We should make sure to unsubscribe at `componentWillUnmount` (Per @calbach's unsubscribe note)

File: ui/src/app/components/help-sidebar.tsx (hunk 5 above)
Comment (dolbeew) [reply ui/src/app/components/help-sidebar.tsx]: Yes, I've changed this after reworking state/props based on your comment below

File: ui/src/app/components/help-sidebar.tsx (hunk 3 above)
Comment (dolbeew) [reply ui/src/app/components/help-sidebar.tsx]: Renamed and changed to boolean along with reworking state/props based on your comment below

File: ui/src/app/pages/data/cohort-review/sidebar-content.component.tsx (hunk 7 above)
Comment (dolbeew) [reply ui/src/app/pages/data/cohort-review/sidebar-content.component.tsx]: Addressed this on the last commit, forgot to comment.
